processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go

```diff
@@ -56,6 +56,11 @@ func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
 		return res, err
 	}
 
+	hostname, err := d.provider.hostname(ctx)
+	if err != nil {
+		return res, err
```

Comment:
Since there are two errors that can be returned now can you add some additional context to both of them so we know which one fails? e.g.

```go
fmt.Errorf("failed getting identity document: %w", err)
```

Reply:
Sure, I added some context to both errors.